src/main/java/me/apisek12/StoneDrop/EventListeners/BlockBreakEventListener.java

```diff
@@ -101,11 +105,38 @@ private static void dropItems(ItemStack itemStack, Player player,  Location loca
             }
         }
         else {
-            location.getWorld().dropItem(location, itemStack);
+            if(PluginMain.dropSpreadMultiDir){
+                chestDropItems(itemStack,player,location);
+            }else {
+                location.getWorld().dropItem(location, itemStack);
+            }
         }
     }
+
+    private static void chestDropItems(ItemStack itemStack, Player player,  Location location){
+        Bukkit.getScheduler().runTaskLater(PluginMain.plugin, () -> {
+            spawnDropChest(itemStack,player,location);
+        }, 4);
+    }
+    private static  void spawnDropChest(ItemStack itemStack, Player player,Location location){
+        location.getBlock().setType(Material.CHEST);
+
+        Chest chest = (Chest) location.getBlock().getState();
+        int chestSlotAmount = (int) (itemStack.getAmount()/27);
+        int extraSlotAmount = (chestSlotAmount < 64) ?  itemStack.getAmount() % 27 : 0;
+        for (int ci=0,extraAdd=0;ci<27;ci++,extraAdd++){
+            int finalSlotItemAmount = (extraAdd < extraSlotAmount ) ? chestSlotAmount + 1 : chestSlotAmount;
+            chest.getInventory().setItem(ci,new ItemStack(itemStack.getType(), finalSlotItemAmount));
+
+        }
+
+        Bukkit.getServer().getPluginManager().callEvent(new BlockBreakEvent(location.getBlock(),player));
```

Comment:
Possible bug when player gets treasure chest spawned at the same time - treasure drop will be overwritten.
This bug should be removed.